lms/djangoapps/courseware/courses.py

```diff
@@ -173,6 +174,30 @@ def can_self_enroll_in_course(course_key):
     return True
 
 
+def course_open_for_self_enrollment(course_key):
+    """
+    For a given course_key, determine if the course is available for enrollment
+    """
+    # Check to see if learners can enroll themselves.
+    if not can_self_enroll_in_course(course_key):
+        return False
+
+    # Check the enrollment start and end dates.
+    course_details = get_course_enrollment_details(unicode(course_key))
+    now = datetime.now().replace(tzinfo=pytz.UTC)
+    start = course_details['enrollment_start']
+    end = course_details['enrollment_end']
+
+    start = start if start is not None else now
+    end = end if end is not None else now
+
+    # If we are not within the start and end date for enrollment.
+    if start > now or now > end:
```

Comment:
No, it's fine. I accidentally screwed it up and put it back together wonky. I don't mind using your version.